src/audio/data_blob.c

```diff
@@ -22,17 +22,40 @@ struct comp_data_blob_handler {
 	uint32_t data_pos;	/**< indicates a data position in data
 				  *  sending/receiving process
 				  */
+	uint32_t single_blob:1; /**< Allocate only one blob. Module can not
+				  *  be active while reconfguring.
+				  */
+	void *(*alloc)(size_t size);	/**< alternate allocator, maybe null */
+	void (*free)(void *buf);	/**< alternate free(), maybe null */
 };
 
+static void *data_blob_alloc(struct comp_data_blob_handler *blob_handler,
+			     size_t size)
+{
+	if (blob_handler->alloc)
+		return blob_handler->alloc(size);
+	else
+		return rballoc(0, SOF_MEM_CAPS_RAM, size);
+}
+
+static void data_blob_free(struct comp_data_blob_handler *blob_handler,
+			   void *buf)
+{
+	if (blob_handler->free)
+		blob_handler->free(buf);
+	else
+		rfree(buf);
+}
```

Comment:
It would be more consistent to check the alloc/free handler once at comp_data_blob_handler_new_full() and if it is NULL set the default handler the call rballoc/rfree.

Reply:
I was trying avoid longish blob_handler->alloc/free() calls around the code, but yes it is rather pointless. I'll go with your suggestion.